internal/parser/parser.go

```diff
@@ -5083,6 +5083,54 @@ func (p *parser) parseStmt(opts parseStmtOpts) ast.Stmt {
 			stmt.IsSingleLine = isSingleLine
 			p.lexer.ExpectContextualKeyword("from")
 
+		case lexer.TTypeof:
+			// "import typeof foo from 'path';"
+			if !p.Flow.Parse {
+				p.lexer.Unexpected()
+				return ast.Stmt{}
+			}
+			p.lexer.Next()
+			hasDefaultSpecifier := false
+			switch p.lexer.Token {
+			case lexer.TIdentifier:
+				// "import typeof defaultName from 'path';"
+				// "import typeof defaultName, {foo, bar} from 'path';"
+				hasDefaultSpecifier = true
+				p.lexer.Next()
+
+			case lexer.TAsterisk:
+				// "import typeof * as foo from 'path';"
+				p.lexer.Next()
+				p.lexer.ExpectContextualKeyword("as")
+				p.lexer.Expect(lexer.TIdentifier)
+
+			case lexer.TOpenBrace:
+				// "import typeof {foo} from 'path';"
+				p.parseImportClause()
+			}
+			if hasDefaultSpecifier && p.lexer.Token == lexer.TComma {
+				p.lexer.Next()
+				switch p.lexer.Token {
+				case lexer.TAsterisk:
+					// "import typeof defaultItem, * as ns from 'path'"
+					p.lexer.Next()
+					p.lexer.ExpectContextualKeyword("as")
+					p.lexer.Expect(lexer.TIdentifier)
+
+				case lexer.TOpenBrace:
+					// "import typeof defaultItem, {item1, item2} from 'path'"
+					p.parseImportClause()
+
+				default:
+					p.lexer.Unexpected()
+				}
+			}
+
+			p.lexer.ExpectContextualKeyword("from")
+			p.parsePath()
+			p.lexer.ExpectOrInsertSemicolon()
+			return ast.Stmt{Loc: loc, Data: &ast.STypeScript{}}
```

Comment:
This is copied from the TypeScript-specific section of the import parser. Should there be a separate AST type for Flow types? Or should should this just emit a blank statement?

Reply:
I think `STypeScript` should be changed to something more generic such as `SIgnoredType`. The only purpose is to tell the caller to drop the statement. These statements never make it out of the parser.